packages/web/docs/src/app/blog/(posts)/introducing-mcp-graphql/page.mdx

```diff
@@ -0,0 +1,89 @@
+
+It does this by allowing AI applications to introspect a local or remote schema and use it as extra context to execute queries and/or mutations on your GraphQL server. There's a single query tool call that allows any kind of operation with any kind of variables to be filled in by the model.
+
+## GraphQL and MCP
+
+GraphQL and MCP are perfectly suited for eachother. Where most MCP servers create specific, narrow tools, with GraphQL the AI gets direct access to the schema and can compose any valid GraphQL operation on the spot.
+
+This means:
+- No need to anticipate every possible use case upfront
+- The AI can combine data in ways you never explicitly programmed
+- Users can ask natural questions without being limited to your predefined interface
+
+In many ways this is exactly what MCP is built for rather than just exposing individual API endpoints.
+
+This way of allowing any kind of operation works great with models as they're quite adept at reading GraphQL schema SDL's and creating operations based on that information. Schema's are fully typed and often well documented, leading to a lot of useful context for an AI to work with.
+
+This also opens up some interesting testing possibilities. Unlike frontend applications that follow predictable query patterns, the AI can explore your entire schema space - making deeply nested queries, testing edge cases, or trying operations that might cause performance issues. You can even prompt it to deliberately search for recursive query vulnerabilities or operations that might cause errors.
+
+## How it works
+- `mcp-graphql` is available as an NPM package
+- You can download and run it using `npx mcp-graphql` where you can use certain environment variables
+- Example for Claude Desktop:
+```json
+{
+    "mcpServers": {
+        "mcp-graphql": {
+            "command": "npx",
+            "args": ["mcp-graphql"],
+            "env": {
+                "ENDPOINT": "http://localhost:3000/graphql"
+            }
+        }
+    }
+}
+```
+
+### What it looks like in practice
+
+Once set up, you can ask natural questions like "Show me users who posted in the last week and their most popular posts." The AI reads your GraphQL schema, understands the relationships between users and posts, and composes something like:
+
+```graphql
+query RecentActiveUsers {
+  users(where: { posts: { some: { createdAt: { gte: "2025-01-19" } } } }) {
+    name
+    email
+    posts(orderBy: { likes: desc }, take: 3, where: { createdAt: { gte: "2025-01-19" } }) {
+      title
+      likes
+      createdAt
+    }
+  }
+}
+```
+
+With traditional MCP servers, this would require separate tools for users, posts, filtering, and sorting - if you even anticipated this exact combination.
+
+## The future
+- Tools per operation
```

Comment:
Turn that into text
also maybe link to PRs or issues on the repo
and expand more on the idea of how to connect to Hive Console